packages/odyssey-react-mui/src/DataTable/DataTable.tsx

```diff
@@ -445,20 +454,51 @@ const DataTable = ({
     ],
   );
 
-  const dataTableFilters = useMemo(
-    () =>
-      columns
-        .filter((column) => column.enableColumnFilter !== false)
-        .map((column) => {
-          return {
-            id: column.accessorKey as string,
-            label: column.header,
-            variant: column.filterVariant ?? "text",
-            options: column.filterSelectOptions,
-          } as DataFilter;
-        }),
-    [columns],
-  );
+  const convertColumnToFilter = (
+    column: DataTableColumn<DataTableRowData>,
+  ): DataFilter | null => {
+    if (!column.enableColumnFilter) {
+      return null;
+    }
+
+    return {
+      id: column.accessorKey,
+      label: column.header,
+      variant: column.filterVariant,
+      options: column.filterSelectOptions,
+    } as DataFilter;
+  };
```

Comment:
I like positive inferences first.

Can I think this could be a ternary implicit return:

```suggestion
  const convertColumnToFilter = (
    column: DataTableColumn<DataTableRowData>,
  ): DataFilter | null => (
    column.enableColumnFilter
    ? {
      id: column.accessorKey,
      label: column.header,
      variant: column.filterVariant,
      options: column.filterSelectOptions,
    }
    : null
  );
```

What do you think?

I also removed the `as DataFilter` in there because you already set the return value. If it doesn't satisfy `DataFilter`, it will give you a type error.

This should also be wrapped in a `useCallback` right?

Reply:
Good call, done 👍